src/filters.rs

```diff
@@ -495,6 +501,92 @@ impl QueryRuntimeFilter for RangeFilter {
     }
 }
 
+pub struct BboxFilter<'a> {
+    snapshot: Rc<Snapshot<'a>>,
+    iter: Option<DBIterator>,
+    kb: KeyBuilder,
+    bbox: [f64; 4],
+    term_ordinal: Option<usize>,
+}
+
+impl<'a> BboxFilter<'a> {
+    pub fn new(snapshot: Rc<Snapshot<'a>>,
+               kb: KeyBuilder,
+               bbox: [f64; 4])
+               -> BboxFilter<'a> {
+        BboxFilter {
+            snapshot: snapshot,
+            iter: None,
+            kb: kb,
+            bbox: bbox,
+            term_ordinal: None,
+        }
+    }
+
+    /// Function to deserialize the Arraypaths
+    fn from_u8_slice(slice: &[u8]) -> Vec<u64> {
+        let u64_slice = unsafe {
+            std::slice::from_raw_parts(slice.as_ptr() as *const u64, slice.len() / 8)
+        };
+        u64_slice.to_vec()
+    }
+}
+
+impl<'a> QueryRuntimeFilter for BboxFilter<'a> {
+    fn first_result(&mut self, start: &DocResult) -> Option<DocResult> {
+        let mut bbox_vec = Vec::new();
```

Comment:
You should put this code that initializes the bbox_vec into the BoxFilter::new(..) method.